### ISSUE TITLE
iRonCub3: smoke from a board on the back 

### ISSUE THREAD

**Issue by gabrielenava** [robotology/icub-tech-support#1579]
### Device name 🤖

iCubGenova09 - iRonCub3

### Request/Failure description

There was smoke and smell of something burning from the upper back of the robot, close to the battery display.

### Detailed context

We connected the robot to the power supply and turned on motors and on board pc. We did not start the yarprobotinterface. After few minutes, we smelled something burning. We saw smoke coming out from the robot upper back (close to the battery display) and we immediately switched off the power.

### Additional context

Blocking iRonCub3 experiments and demos cc @DanielePucci

**Comment by sgiraz** [comment]
Hi @gabrielenava,

In accordance with @maggia80, we are going to check the status of the robot, in particular the battery and the related electronics with @MrAndrea.

cc @AntonioAzocar @Gandoo @Fabrizio69

**Comment by MrAndrea** [comment]
I checked the robot together with @Gandoo.
The problem was with the 5V regulator on the BAT board, it provides... 20V instead of 5V. The smoke comes from the unfortunate display on the back, which is the only device connected to that regulator.
I don't know the cause of this, maybe a dropped screw or debris that got inside the covers?
There are no other problems.

Therefore, the solution is to replace the BAT board (iitcode 10050.D) and the display (iitcode 7282) with the microSD (iitcode 15022).

Note:
- before mounting the new BAT, I have to check if the firmware revision is the 1.2.0 (see [this PR](https://github.com/robotology/icub-firmware-build/pull/86))
- the new display must be programmed

**Comment by gabrielenava** [comment]
thank you @MrAndrea!

**Comment by MrAndrea** [comment]
I programmed a new BAT board and the display. We can proceed with the replacement.

cc @Gandoo @Fabrizio69

**Comment by MrAndrea** [comment]
Together with @Gandoo we replaced the faulty boards. 
Power tests were done (both battery and power supply) and were successful, the robot now works again.

cc @Fabrizio69 @maggia80

**Comment by gabrielenava** [comment]
today afternoon me and @vpunithreddy tried to start the robot, bu there were issues: see https://github.com/robotology/icub-tech-support/issues/1581

I don't think the problem is correlated, but it needs to be fixed in order to complete the testing. 

Anyways we also connected the robot to the power supply and there was no smoke or burning smell.

A question for @MrAndrea: we are using these settings with our power supply:

![photo1687281561](https://github.com/robotology/icub-tech-support/assets/12396934/33de1383-9975-450d-8f74-f7a44fea7f10)

we never changed them since we got the power supply, are they the correct ones for iCub3?
